var/spack/repos/builtin/packages/elk/package.py

```diff
@@ -29,61 +36,66 @@ class Elk(Package):
     # Cannot be built in parallel
     parallel = False
 
-
     def configure(self, spec):
         # Dictionary of configuration options
         config = {
-            'MAKE':      'make',
-            'F90':       join_path(spack.build_env_path, 'f90'),
-            'F77':       join_path(spack.build_env_path, 'f77'),
-            'AR':        'ar',
-            'LIB_FFT':   'fftlib.a',
-            'SRC_MPI':   'mpi_stub.f90',
-            'SRC_OMP':   'omp_stub.f90',
-            'SRC_libxc': 'libxcifc_stub.f90',
-            'SRC_FFT':   'zfftifc.f90'
+            'MAKE': 'make',
+            'AR':   'ar'
         }
 
         # Compiler-specific flags
         flags = ''
-        if   self.compiler.name == 'intel':
-            flags = '-O3 -ip -unroll -no-prec-div -openmp'
+        if self.compiler.name == 'intel':
+            flags = '-O3 -ip -unroll -no-prec-div'
         elif self.compiler.name == 'gcc':
-            flags = '-O3 -ffast-math -funroll-loops -fopenmp'
+            flags = '-O3 -ffast-math -funroll-loops'
         elif self.compiler.name == 'pgi':
-            flags = '-O3 -mp -lpthread'
+            flags = '-O3 -lpthread'
         elif self.compiler.name == 'g95':
             flags = '-O3 -fno-second-underscore'
         elif self.compiler.name == 'nag':
             flags = '-O4 -kind=byte -dusty -dcfuns'
         elif self.compiler.name == 'xl':
-            flags = '-O3 -qsmp=omp'
+            flags = '-O3'
         config['F90_OPTS'] = flags
         config['F77_OPTS'] = flags
 
         # BLAS/LAPACK support
+        # Note: BLAS/LAPACK must be compiled with OpenMP support
+        # if the +openmp variant is chosen
         blas   = 'blas.a'
         lapack = 'lapack.a'
-        if '+blas'   in spec:
-            blas   = join_path(spec['blas'].prefix.lib,   'libblas.so')
+        if '+blas' in spec:
+            blas   = spec['blas'].blas_shared_lib
         if '+lapack' in spec:
-            lapack = join_path(spec['lapack'].prefix.lib, 'liblapack.so')
-        config['LIB_LPK'] = ' '.join([lapack, blas]) # lapack must come before blas
+            lapack = spec['lapack'].lapack_shared_lib
+        # lapack must come before blas
```

Comment:
> Of course it depends on what the package does with it, but normally one would have linking flag -lblas -llapack. On another hand modern compilers can search for symbols both ways.

I think this is wrong. The linker usually inspects libraries from left to right, and tries to resolve symbols on the way. The right order is:
```
-llapack -lblas
```
because lapack has unresolved symbols for blas. See [here](http://stackoverflow.com/questions/45135/why-does-the-order-in-which-libraries-are-linked-sometimes-cause-errors-in-gcc)